lms/views/decorators/h_api.py

```diff
@@ -106,6 +111,41 @@ def wrapper(context, request):
     return wrapper
 
 
+def upsert_group_info(wrapped):
+    """Create or update the :class:`lms.models.GroupInfo` for this launch."""
+
+    @functools.wraps(wrapped)
+    def wrapper(context, request):
+        if not context.provisioning_enabled:
+            return wrapped(context, request)
+
+        request.find_service(name="group_info_upsert")(
+            context.h_authority_provided_id,
+            request.lti_user.oauth_consumer_key,
+            **{
+                param: request.params.get(param)
+                for param in [
+                    "context_id",
+                    "context_title",
+                    "context_label",
+                    "tool_consumer_info_product_family_code",
+                    "tool_consumer_info_version",
+                    "tool_consumer_instance_name",
+                    "tool_consumer_instance_description",
+                    "tool_consumer_instance_url",
+                    "tool_consumer_instance_contact_email",
+                    "tool_consumer_instance_guid",
+                    "custom_canvas_api_domain",
+                    "custom_canvas_course_id",
+                ]
+            },
```

Comment:
This is a dict comprehension that pulls a bunch of params out of the request (or `None` for each param that doesn't exist). This sort of thing belongs in the `lms/validation/` package (and written as a Marshmallow schema) but in this case I judged it not worth the trouble

Reply:
Truth be told, we likely don't care much about the exact contents here. Only if something causes an exception. In this case the worse that should happen is you get None if they changed the names.